src/riscv/lib/src/pvm/linux/signals.rs

```diff
@@ -23,6 +23,87 @@ pub const SIZE_SIGACTION: usize = 32;
 /// As we're building a 64-bit system, the sigset should be 64-bit wide as well.
 pub const SIGSET_SIZE: u64 = 8;
 
+/// Linux signal signums in RISC-V, see <https://www.man7.org/linux/man-pages/man7/signal.7.html>
+#[derive(Debug, Clone, Copy)]
+pub enum Signal {
+    Sigill = 4,
+    Sigabrt = 5,
+    Sigiot = 6,
+    Sigbus = 7,
+    Sigfpe = 8,
+    Sigkill = 9,
+    Sigusr1 = 10,
+    Sigsegv = 11,
+    Sigusr2 = 12,
+    Sigpipe = 13,
+    Sigterm = 15,
+    Sigstop = 19,
+    Sigsys = 31,
+}
+
+impl TryFrom<u64> for Signal {
+    type Error = Error;
+
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        match value {
+            4 => Ok(Signal::Sigill),
+            5 => Ok(Signal::Sigabrt),
+            6 => Ok(Signal::Sigiot),
+            7 => Ok(Signal::Sigbus),
+            8 => Ok(Signal::Sigfpe),
+            9 => Ok(Signal::Sigkill),
+            10 => Ok(Signal::Sigusr1),
+            11 => Ok(Signal::Sigsegv),
+            12 => Ok(Signal::Sigusr2),
+            13 => Ok(Signal::Sigpipe),
+            15 => Ok(Signal::Sigterm),
+            19 => Ok(Signal::Sigstop),
+            31 => Ok(Signal::Sigsys),
+            _ => Err(Error::InvalidArgument),
+        }
```

Comment:
Would something like this work?


```suggestion
        let repr_value: u8 = value.try_into().map_err(|_| Error::InvalidArgument)?;
        Self::from_repr(repr_value).ok_or(Error::InvalidArgument)
```

Reply:
As you're on leave, I'll resolve these myself.